filemanager.lua

```diff
@@ -180,39 +187,36 @@ local function get_scanlist(dir, ownership, indent_n)
 	-- Hold the current scan's filename in most of the loops below
 	local filename
 
-	-- Splitting the loops for speed, so we don't run an unnecessary if every pass
-	if not show_dotfiles and not show_ignored then
-		-- Don't show dotfiles or ignored
-		for i = 1, #dir_scan do
-			filename = dir_scan[i]:Name()
-			-- Check if it's a hidden file
-			if not is_dotfile(filename) and not is_ignored_file(filename) then
-				-- Since we skip indicies of dotfiles, don't use i here or we add nil values
-				results[#results + 1] = get_results_object(filename)
-			end
+	for i = 1, #dir_scan do
+    	local showfile = true
+		filename = dir_scan[i]:Name()
+		-- If we should not show dotfiles, and this is a dotfile, don't show
+		if not show_dotfiles and is_dotfile(filename) then
+			showfile = false
 		end
-	elseif show_dotfiles and not show_ignored then
-		-- Show dotfiles but not ignored
-		for i = 1, #dir_scan do
-			filename = dir_scan[i]:Name()
-			if not is_ignored_file(filename) then
-				results[#results + 1] = get_results_object(filename)
-			end
+		-- If we should not show ignored files, and this is an ignored file, don't show
+		if not show_ignored and is_ignored_file(filename) then
+			showfile = false
 		end
-	elseif not show_dotfiles and show_ignored then
-		-- Show ignored but not dotfiles
-		for i = 1, #dir_scan do
-			filename = dir_scan[i]:Name()
-			if not is_dotfile(filename) then
+		if showfile then
+		    -- This file is good to show, proceed
+			if folders_first and not is_dir(JoinPaths(dir,filename)) then
+			    -- If folders_first and this is a file, add it to (temporary) files
+				files[#files + 1] = get_results_object(filename)
+			else
+    		    -- Otherwise, add to results
 				results[#results + 1] = get_results_object(filename)
 			end
 		end
-	else
-		-- Show dotfiles and ignored (aka everything)
-		for i = 1, #dir_scan do
-			results[i] = get_results_object(dir_scan[i]:Name())
+	end
+	if #files > 0 then
+	    -- Append any files to results, now that all folders have been added
+	    -- files will be > 0 only if folders_first and there are files
+		for i = 0, #files do
```

Comment:
Index starts at 1 in Lua (unless you specifically create a zero index). This is (probably) what's causing the issues with `try_open_at_y`, as when it goes to access it, it gets a `nil` value.

There might even be a built-in "append table" type function in the Lua library, although I haven't looked at the reference manual in a while.